### PR TITLE
Adding Min/Max support for decompogen.py

### LINKED ISSUE #13612: decompogen doesn't like Min/Max
```
>>> decompogen(Min(5,x),x)
...
  File "sympy\solvers\decompogen.py", line 57, in decompogen
    result += [f1] + decompogen(f2, symbol)
  File "sympy\solvers\decompogen.py", line 57, in decompogen
    result += [f1] + decompogen(f2, symbol)
  File "sympy\solvers\decompogen.py", line 57, in decompogen
    result += [f1] + decompogen(f2, symbol)
  File "sympy\solvers\decompogen.py", line 51, in decompogen
    fp = Poly(f)
  File "sympy\polys\polytools.py", line 128, in __new__
    return cls._from_expr(rep, opt)
  File "sympy\polys\polytools.py", line 238, in _from_expr
    rep, opt = _dict_from_expr(rep, opt)
  File "sympy\polys\polyutils.py", line 367, in _dict_from_expr
    rep, gens = _dict_from_expr_no_gens(expr, opt)
  File "sympy\polys\polyutils.py", line 310, in _dict_from_expr_no_gens
    (poly,), gens = _parallel_dict_from_expr_no_gens((expr,), opt)
  File "sympy\polys\polyutils.py", line 273, in _parallel_dict_from_expr_no_gens

    gens = _sort_gens(gens, opt=opt)
  File "sympy\polys\polyutils.py", line 106, in _sort_gens
    gens = sorted(gens, key=order_key)
  File "sympy\polys\polyutils.py", line 78, in order_key
    gen = str(gen)
  File "sympy\core\basic.py", line 400, in __str__
    return sstr(self, order=None)
  File "sympy\printing\str.py", line 796, in sstr
    s = p.doprint(expr)
  File "sympy\printing\printer.py", line 233, in doprint
    return self._str(self._print(expr))
  File "sympy\printing\printer.py", line 257, in _print
    return getattr(self, printmethod)(expr, *args, **kwargs)
  File "sympy\printing\str.py", line 210, in _print_LatticeOp
    args = sorted(expr.args, key=default_sort_key)
  File "sympy\core\compatibility.py", line 462, in default_sort_key
    return item.sort_key(order=order)
  File "sympy\core\cache.py", line 93, in wrapper
    retval = cfunc(*args, **kwargs)
  File "sympy\core\compatibility.py", line 792, in wrapper
    key = make_key(args, kwds, typed) if kwds or typed else args
  File "sympy\core\compatibility.py", line 724, in _make_key
    return _HashedSeq(key)
  File "sympy\core\compatibility.py", line 702, in __init__
    self.hashvalue = hash(tup)
RuntimeError: maximum recursion depth exceeded
```

perhaps related to #13205 and #13207

### PR DESCRIPTION
**Pull request by prajwalgatti**
Fixes #13612 

Now the code produces the following output:
`>>>decompogen(Max(x, 3), x)`
  `[Max]`
 `>>>decompogen(Max(Min(sin(tan(x**4)), pi), sin(tan(x))), x)`
  `[Max, sin(x), tan(x), Min, sin(x), tan(x), x**4] `

### CODE REVIEW

**Comment by prajwalgatti** [comment]
@smichr, please review

**Comment by smichr** [comment]
ping @aktech , can you take a look at this? It seems too fragile to only be traversing certain args. Perhaps you could make a better suggestion.

File: sympy/solvers/decompogen.py [hunk 1]
```diff
@@ -47,6 +50,14 @@ def decompogen(f, symbol):
         result += [f.subs(f.args[0], symbol)] + decompogen(f.args[0], symbol)
         return result
 
+    # ===== Max/Min Functions ===== #
+    if isinstance(f, (Max, Min)):
+        result += [type(f)]
```
Comment (smichr) [inline sympy/solvers/decompogen.py]: Adding what you are going to traverse in the `for` loop makes this fragile -- trust recursion here:
```
if symbol not in f.free_symbols or f == symbol:
    return [f]

if isinstance(f, (Function, Pow)):
...
if isinstance(f, (Max, Min)):
    rv = [type(f)]
    for i in f.args:
        rv.extend(decompogen(i, symbol))
    return rv
```

But I don't know what is going to be done with the results and whether it is important to know that Min and Max were present...so maybe it should start out as `rv = []`.

**Comment by czgdp1807** [comment]
@prajwalgatti Are you still interested to work on it?

**Pull request closed without merging**